### PR TITLE
add support for user impersonation for k8s modules

### LINKED ISSUE #40: impersonation support using become in k8s
##### SUMMARY
I'm only using Openshift and never used k8s
One of the nifty feature is to be able to impersonate users so I can do tasks as my personal user to work which have extended privileges (but not admin) and for some tasks I can use oc --as system:admin to perform some as cluster-admin role.

This is not exposed in the k8s module, which could be helpful, expose as a `become` keyword

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
k8s

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

k8s module should not allow user to perform operation using impersonation as describe here 
https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation
This pull request closes #40 

##### ISSUE TYPE

- Feature Pull Request
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

File: plugins/doc_fragments/k8s_auth_options.py [hunk 1]
```diff
@@ -119,6 +119,19 @@ class ModuleDocFragment(object):
...
+  impersonate_user:
+    description:
+    - Username to impersonate for the operation.
+    - Can also be specified via K8S_AUTH_IMPERSONATE_USER environment.
+    type: str
+    version_added: 2.3.0
+  impersonate_groups:
+    description:
+    - Group(s) to impersonate for the operation.
+    - Can also be specified via K8S_AUTH_IMPERSONATE_GROUPS environment. Example: 'Group1,Group2'
```
Comment (gravesm) [inline plugins/doc_fragments/k8s_auth_options.py:131]:
```suggestion
    - "Can also be specified via K8S_AUTH_IMPERSONATE_GROUPS environment. Example: 'Group1,Group2'"
```

**Review by gravesm — APPROVED** [review]
Nice!

**Review by Akasurde — APPROVED** [review]
LGTM. Great work @abikouo

**Pull request merged**